native/src/main/java/io/ballerina/stdlib/http/api/HttpInterceptorUnitCallback.java

```diff
@@ -160,7 +162,16 @@ public void notifyFailure(BError result) {
                 sendFailureResponse(result);
             }
         };
-        runtime.invokeMethodAsync(caller, "returnResponse", null,
-                ModuleUtils.getNotifySuccessMetaData(), returnCallback, paramFeed);
+        ObjectType callerType = caller.getType();
+        String remoteMethod = "returnResponse";
+        if (callerType.isIsolated() && callerType.isIsolated(remoteMethod)) {
```

Comment:
`returnResponse` is some internal method, since we make the caller Isolated, and `returnResponse` is already isolated,  this condition will be always true. I think we can only call `invokeMethodAsyncConcurrently` here

Reply:
Fixed with [4d461ab74ee46c7a163d480ff8827652e794cf51](https://github.com/ballerina-platform/module-ballerina-http/pull/765/commits/4d461ab74ee46c7a163d480ff8827652e794cf51)